### PR TITLE
Add a tutorial for plotting focal mechanisms (beachballs)

### PR DESCRIPTION
**Pull request by yvonnefroehlich**
**Description of proposed changes**

This PR aims to add a tutorial showing how to plot focal mechanisms using `pygmt.Figure.meca`. It focuses on how the display of the beachballs can be adjusted and modified:
- Filling the quadrants
- ~Plotting the P and T axes~
- Plotting the components of a seismic moment tensor
- Adjusting the outlines
- Highlighting the nodal planes
- Adding offset from the event location
- Adding a label
- Using size-coding and color-coding

**Related PRs**  (need to be merged before this PR is mergd)

- [x] Add aliases for **L**, **T**, **Fr** :  PR #2546 
- [x] Expand docstring for ``scale``: PR #2552 
- [x] Expand docstrings for ``compressionfill``, ``camp``, ``offset``: PR #2553 
- [x] Add table for conventions to docs: PR #3811 
- [ ] ~Add aliases for **Fa**, **Fe**, **Fg**, **Ft**, **Fp**: PR #3526~

**Backup of code for plotting the P and T axes**
```python
# %%
# Plotting the P and T axes
# -------------------------
#
# Wait for PR #3526

fig = pygmt.Figure()
fig.basemap(region=[-size, size] * 2, projection=projection, frame=frame)

fig.meca(
    spec=mt_dict_single,
    scale="1c",
    longitude=-2,
    latitude=0,
    Fa=True,
)

fig.meca(
    spec=mt_dict_single,
    scale="1c",
    longitude=2,
    latitude=0,
    Fa="0.2c/cd",  # Compute and plot P and T axes with symbols, Adjust size and symbols
    Fe="darkorange",  # Adjust fill of T axis symbol
    Fg="gray30",  # Adjust fill of P axis symbol
    Ft="0.8p,cornsilk",  # Adjust outline of T axis symbol
    Fp="0.8p,gray60",  # Adjust outline of P axis symbol
)

fig.show()
```

**Preview**: https://pygmt-dev--2550.org.readthedocs.build/en/2550/tutorials/advanced/focal_mechanisms.html

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.
- [ ] Use underscores (not hyphens) in names of Python files and directories.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

File: examples/tutorials/advanced/focal_mechanisms.py [hunk 1]
```diff
@@ -0,0 +1,449 @@
...
+# Plot focal mechanism
+fig.meca(
+    spec=fm_collection,
+    scale="0.3c",
+    cmap=True,
+    outline=True,
+)
+
+# Add colorbar
+fig.colorbar(frame=["x+lhpyocentral depth", "y+lkm"])
```
Comment (michaelgrund) [inline examples/tutorials/advanced/focal_mechanisms.py:413]:
```suggestion
fig.colorbar(frame=["x+lhypocentral depth", "y+lkm"])
```

File: examples/tutorials/advanced/focal_mechanisms.py [hunk 2]
```diff
@@ -0,0 +1,449 @@
...
+    frame=["af", "WSne"],
+)
+
+# Plot focal mechanism
+fig.meca(
+    spec=fm_single,
+    scale="1c",  # in centimeters
+    longitude=-79.611,  # event longitude
+    latitude=0.904,  # event latitude
+    pen="1p,blue,solid",
```
Comment (michaelgrund) [inline examples/tutorials/advanced/focal_mechanisms.py:125]:
```suggestion
    pen="1p,gold,solid",
```
Would suggest to use another (lighter) color than default value, makes it easier to distinguish the outline from the black fill.

Comment (yvonnefroehlich) [reply examples/tutorials/advanced/focal_mechanisms.py:125]: Thanks @michaelgrund for looking at this PR at this early stage. I will consider your review when working on this tutorial again. However, I will not be able to finish it in the next few days to get it into the v.0.10.0 release.

**Comment by yvonnefroehlich** [comment]
So far this tutorial covers these aspects:

- Filling the quadrants
- Plotting the components of a seismic moment tensor (maybe to specific ?)
- Adjusting the outlines
- Highlighting the nodal planes
- Adding offset from the event location
- Adding a label
- Using size-coding and color-coding

It does not address details regarding allowed input formats and supported conventions.

Are there any other aspects people think are important to mention? If not, I will continue to improve and complete the comments and documentation 🙂.

**Comment by seisman** [comment]
It's unlikely we can finish PR #3526 before v0.14.0, so remove it from the v0.14.0 milestone.

File: examples/tutorials/advanced/focal_mechanisms.py [hunk 3]
```diff
@@ -0,0 +1,369 @@
...
+    longitude=2,
+    latitude=0,
+    Fa="0.2c/cd",  # Compute and plot P and T axes with symbols, Adjust size and symbols
+    Fe="darkorange",  # Adjust fill of T axis symbol
+    Fg="gray30",  # Adjust fill of P axis symbol
+    Ft="0.8p,cornsilk",  # Adjust outline of T axis symbol
+    Fp="0.8p,gray60",  # Adjust outline of P axis symbol
+)
+
+fig.show()
```
[the remark this answers is not in the document]
Comment (yvonnefroehlich) [reply examples/tutorials/advanced/focal_mechanisms.py]: Yeah, also was having this in my mind as an option to get this tutorial hopefully finished for v0.15.0 (the PR was started in 2023 🙁). Done in commit https://github.com/GenericMappingTools/pygmt/pull/2550/commits/e5151ffba05db6471258279d6bb995c261ffeba1.

File: examples/tutorials/advanced/focal_mechanisms.py [hunk 4]
```diff
@@ -0,0 +1,336 @@
+"""
+Plotting focal mechanisms
+=========================
+
+Focal mechanisms can be plotted as beachballs with the :meth:`pygmt.Figure.meca`
+method.
+
+The focal mechanism data or parameters can be provided as different input types:
+external file, 1-D or 2-D ``numpy.array``, dictionary, ``pandas.Dataframe``.
```
Comment (seisman) [inline examples/tutorials/advanced/focal_mechanisms.py:9]:
```suggestion
an ASCII file, 1-D or 2-D :class:`numpy.array`, dictionary, :class:`pandas.Dataframe`.
```

File: examples/tutorials/advanced/focal_mechanisms.py [hunk 5]
```diff
@@ -0,0 +1,336 @@
...
+Different conventions to define the focal mechanism are supported: Aki and Richards
+(``"aki"``), global CMT (``"gcmt"``), moment tensor (``"mt"``), partial focal
+mechanism (``"partial"``), principal axis (``"principal_axis"``). Please refer to
```
Comment (seisman) [inline examples/tutorials/advanced/focal_mechanisms.py:12]:
```suggestion
mechanism (``"partial"``), and, principal axis (``"principal_axis"``). Please refer to
```

File: examples/tutorials/advanced/focal_mechanisms.py [hunk 6]
```diff
@@ -0,0 +1,336 @@
...
+the documentation of :meth:`pygmt.Figure.meca` regarding how to set up the input
+data in respect to the chosen input type and convention (i.e the expected column
```
Comment (seisman) [inline examples/tutorials/advanced/focal_mechanisms.py:14]:
```suggestion
data in respect to the chosen input type and convention (i.e., the expected column
```

File: examples/tutorials/advanced/focal_mechanisms.py [hunk 7]
```diff
@@ -0,0 +1,336 @@
...
+
+
+# %%
+# Set up the focal mechanism data
+# -------------------------------
+#
+# Store focal mechanism parameters for one event in a dictionary following the
+
+# moment tensor convention
+mt_dict_single = mt_virginia = {
```
Comment (seisman) [inline examples/tutorials/advanced/focal_mechanisms.py:45]:
```suggestion
mt_dict_single = {
```

File: examples/tutorials/advanced/focal_mechanisms.py [hunk 8]
```diff
@@ -0,0 +1,336 @@
...
+- Adding a label
+- Using size-coding and color-coding
+"""
+
+# %%
+import pandas as pd
+import pygmt
+
+# Set up arguments for basemap
+size = 5
```
Comment (seisman) [inline examples/tutorials/advanced/focal_mechanisms.py:33]: Instead of defining `size` and using `region=[-size, size] * 2`, maybe just define `region=[-5, 5, -5, 5]`?

File: examples/tutorials/advanced/focal_mechanisms.py [hunk 9]
```diff
@@ -0,0 +1,336 @@
...
+frame = ["af", "+ggray90"]
+
+
+# %%
+# Set up the focal mechanism data
+# -------------------------------
+#
+# Store focal mechanism parameters for one event in a dictionary following the
+
+# moment tensor convention
```
Comment (seisman) [inline examples/tutorials/advanced/focal_mechanisms.py:44]: Any specific reason that you need to define two focal mechanims here?

Comment (yvonnefroehlich) [reply examples/tutorials/advanced/focal_mechanisms.py:44]: Hm. I think I wanted to have the moment tensor convention for section "Plotting the components of a seismic moment tensor" (around line 109) and later the Aki and Richards convention for the sections "Adjusting the outlines" (around line 143) and "Highlighting the nodal planes" (around line 173).

File: examples/tutorials/advanced/focal_mechanisms.py [hunk 10]
```diff
@@ -0,0 +1,336 @@
...
+    scale="1c",
+    longitude=3,
+    latitude=0,
+    plot_longitude=1,
+    plot_latitude=2,
+    offset="+p1p,darkorange+s0.25c",
+    compressionfill="lightorange",
+)
+
+fig.show()
```
Comment (yvonnefroehlich) [inline examples/tutorials/advanced/focal_mechanisms.py:256]: Looking at the preview at https://pygmt-dev--2550.org.readthedocs.build/en/2550/tutorials/advanced/focal_mechanisms.html#adding-offset-from-event-location, there seem to be a new issue with the offset when using GMT 6.5.0 and PyGMT dev / v0.14.2. I can reproduce this issue locally (under Windows), but on Linux using GMT 6.5.0 and PyGMT v0.14.0. is works as expected:

| GMT dev / 6.5.0 and PyGMT dev / v0.14.2 | GMT 6.5.0 and PyGMT v0.14.0 |
| --- | --- |
| meca_offset_gmt650_pygmt0142 | meca_offset_gmt650_pygmt0140 |

File: examples/tutorials/advanced/focal_mechanisms.py (hunk 8 above)
Comment (yvonnefroehlich) [reply examples/tutorials/advanced/focal_mechanisms.py]: You are right, this is easier; updated in commit https://github.com/GenericMappingTools/pygmt/pull/2550/commits/34c392c74552451e2c96a43cb1641c36aaed9b7d.

File: examples/tutorials/advanced/focal_mechanisms.py (hunk 10 above)
Comment (seisman) [reply examples/tutorials/advanced/focal_mechanisms.py]: The preview shows beachballs with offsetting. Am I misunderstanding here?

File: examples/tutorials/advanced/focal_mechanisms.py (hunk 9 above)
Comment (yvonnefroehlich) [reply examples/tutorials/advanced/focal_mechanisms.py:44]: Hm. I think I wanted to have the moment tensor convention for section "Plotting the components of a seismic moment tensor" (around line 109) and later the Aki and Richards convention for the sections "Adjusting the outlines" (around line 143) and "Highlighting the nodal planes" (around line 173).

File: examples/tutorials/advanced/focal_mechanisms.py (hunk 10 above)
Comment (yvonnefroehlich) [reply examples/tutorials/advanced/focal_mechanisms.py]: The image in the left column of the table is the image currently shown in the preview of this PR. There the beachballs are plotted at the positions given by `longitude` and `latitude`. But actually, the beachballs should be plotted at the positions given by `plot_longitude` and `plot_latitude`, and there should be a line and a small circle highlighting the even location. This is shown in the image in the right column of the table. For me, it works with v0.14.0, but not with v0.14.2.

Comment (seisman) [reply examples/tutorials/advanced/focal_mechanisms.py]: > The image in the left column of the table is the image currently shown in the preview of this PR.

Hmm, I still see the right figure in the preview docs. I also tried it locally with PyGMT dev, and got the right one.

File: examples/tutorials/advanced/focal_mechanisms.py (hunk 9 above)
Comment (seisman) [reply examples/tutorials/advanced/focal_mechanisms.py:44]: OK, I see your points. What about renaming them to shorter names like `meca_aki`/`meca_mt`?

File: examples/tutorials/advanced/focal_mechanisms.py (hunk 10 above)
Comment (yvonnefroehlich) [reply examples/tutorials/advanced/focal_mechanisms.py]: Hm. Somehow wired, when using another browser, the preview looks correct. Anyways, seems like the issue is on my side, so this should be OK 🙂.

File: examples/tutorials/advanced/focal_mechanisms.py [hunk 11]
```diff
@@ -0,0 +1,351 @@
...
+region = [-5, 5, -5, 5]
+projection = "X10c/4c"
+frame = ["af", "+ggray90"]
+
+
+# %%
+# Set up the focal mechanism data
+# -------------------------------
+#
+# Store focal mechanism parameters for one event in a dictionary following the
```
Comment (seisman) [inline examples/tutorials/advanced/focal_mechanisms.py:42]:
```suggestion
# Here we store focal mechanism parameters (in moment tensor and Aki & Richards convention)
for one event in a dictionary:
```

File: examples/tutorials/advanced/focal_mechanisms.py [hunk 12]
```diff
@@ -0,0 +1,351 @@
...
+}
+# Aki and Richards convention
+aki_dict_single = {"strike": 318, "dip": 89, "rake": -179, "magnitude": 7.75}
+
+
+# %%
+# Plotting a single beachball
+# ---------------------------
+#
+# Required parameters are ``spec` and ``scale`` as well as ``longitude``, ``latitude``
```
Comment (seisman) [inline examples/tutorials/advanced/focal_mechanisms.py:62]:
```suggestion
# Required parameters are ``spec`` and ``scale`` as well as ``longitude``, ``latitude``
```

File: examples/tutorials/advanced/focal_mechanisms.py [hunk 13]
```diff
@@ -0,0 +1,351 @@
...
+# (event location), and depth (if these values are not included in the argument passed
+# to ``spec``). Additionally the ``convention`` parameter is required if ``spec`` is
+# an 1-D or 2-D numpy array; for the input types dictionary and ``pandas.Dataframe``,
+# ``convention`` is not needed.
```
Comment (seisman) [inline examples/tutorials/advanced/focal_mechanisms.py:66]:
```suggestion
# the focal mechanism convention is automatically determined from dictionary keys or 
:class:`pandas.DataFrame` column names.
```

File: examples/tutorials/advanced/focal_mechanisms.py (hunk 12 above)
Comment (seisman) [inline examples/tutorials/advanced/focal_mechanisms.py:62]: Need to mention that `scale` controls the size of beachballs.

File: examples/tutorials/advanced/focal_mechanisms.py [hunk 14]
```diff
@@ -0,0 +1,351 @@
...
+    outline=True,
+)
+
+fig.show()
+
+
+# %%
+# Plotting the components of a seismic moment tensor
+# --------------------------------------------------
+#
```
Comment (seisman) [inline examples/tutorials/advanced/focal_mechanisms.py:115]: What about moving this section before the "Filling the quadrants" section? I.e., plot the expected beachballs first before tweaking the apperance.

File: examples/tutorials/advanced/focal_mechanisms.py [hunk 15]
```diff
@@ -0,0 +1,351 @@
...
+# Use the ``component`` parameter to plot the components of a seismic moment tensor.
```
Comment (seisman) [inline examples/tutorials/advanced/focal_mechanisms.py:116]: Maybe rephrase like below? Feel free to make any changes:
```suggestion
# A moment tensor can be decomposed into isotropic and deviatoric parts, and deviatoric 
# part can be further decomposed into multiple parts (e.g., a double couple and A compensated linear vector dipole (CLVD)).
# Use the ``component`` parameter to specify the component to plot.
```

File: examples/tutorials/advanced/focal_mechanisms.py (hunk 12 above)
Comment (yvonnefroehlich) [reply examples/tutorials/advanced/focal_mechanisms.py]: Please see commit https://github.com/GenericMappingTools/pygmt/pull/2550/commits/350d01d2cf2066d79d7f7651a99252099df515ff for explaining `scale`.

File: examples/tutorials/advanced/focal_mechanisms.py (hunk 14 above)
Comment (yvonnefroehlich) [reply examples/tutorials/advanced/focal_mechanisms.py]: There was something I had in mind regarding the order of the sections; however, I am currently not remembering my idea here 🙈 (maybe this was somehow related to the example for **Fa**, **Fe**, **Fg**, **Ft**, **Fp**). As I think your suggestion makes sense, I move the "Filling the quadrants" section up in commit https://github.com/GenericMappingTools/pygmt/pull/2550/commits/57121bfe702b4f1237317878b8278b454ad9653d.

File: examples/tutorials/advanced/focal_mechanisms.py (hunk 9 above)
Comment (yvonnefroehlich) [reply examples/tutorials/advanced/focal_mechanisms.py:45]: My idea of the variable names was ``convention_datatype_amount``:
- aki_dict_single
- mt_dict_single
- aki_dict_multiple
- aki_df_multiple

File: examples/tutorials/advanced/focal_mechanisms.py [hunk 16]
```diff
@@ -0,0 +1,359 @@
...
+    "plot_latitude": [40, 10, 50, 55],
+    "event_name": [
+        "Haiti - 2010/01/12",
+        "Esmeraldas - 2022/03/27",
+        "Afghanistan - 2022/06/21",
+        "Syria/Turkey - 2023/02/06",
+    ],
+}
+# Convert to a pandas.DataFrame
+aki_df_multiple = pd.DataFrame(aki_dict_multiple)
```
Comment (seisman) [inline examples/tutorials/advanced/focal_mechanisms.py:308]: `aki_df_dict` is not used:
```suggestion
# Set up a pandas.DataFrame with multiple focal mechanism parameters.
aki_df_multiple = pd.DataFrame(
    {
        "strike": [255, 173, 295, 318],
        "dip": [70, 68, 79, 89],
        "rake": [20, 83, -177, -179],
        "magnitude": [7.0, 5.8, 6.0, 7.8],
        "longitude": [-72.53, -79.61, 69.46, 37.01],
        "latitude": [18.44, 0.90, 33.02, 37.23],
        "depth": [13, 19, 4, 10],
        "plot_longitude": [-70, -110, 100, 0],
        "plot_latitude": [40, 10, 50, 55],
        "event_name": [
            "Haiti - 2010/01/12",
            "Esmeraldas - 2022/03/27",
            "Afghanistan - 2022/06/21",
            "Syria/Turkey - 2023/02/06",
        ],
    }
)
```

File: examples/tutorials/advanced/focal_mechanisms.py [hunk 17]
```diff
@@ -0,0 +1,359 @@
...
+
+fig.meca(
+    spec=aki_dict_single,
+    scale="1c",
+    longitude=-1,
+    latitude=0,
+    depth=0,
+    plot_longitude=-3,
+    plot_latitude=2,
+    offset=True,
```
Comment (seisman) [inline examples/tutorials/advanced/focal_mechanisms.py:264]: `offset` is automatically set when `plot_longitude`/`plot_latitude` is given.
```suggestion
```

File: examples/tutorials/advanced/focal_mechanisms.py [hunk 18]
```diff
@@ -0,0 +1,359 @@
...
+
+fig.show()
+
+
+# %%
+# Plotting multiple beachballs
+# ----------------------------
+#
+# Data of four earthquakes taken from USGS.
+# Provide lists.
```
Comment (seisman) [inline examples/tutorials/advanced/focal_mechanisms.py:287]: Need changes.

File: examples/tutorials/advanced/focal_mechanisms.py [hunk 19]
```diff
@@ -0,0 +1,359 @@
...
+    component="dc",  # closest double couple
+)
+fig.meca(
+    spec=mt_dict_single,
+    scale="1c",
+    longitude=2,
+    latitude=0,
+    depth=0,
+    component="deviatoric",  # deviatoric part
+)
```
[the remark this answers is not in the document]
Comment (yvonnefroehlich) [reply examples/tutorials/advanced/focal_mechanisms.py]: Changed the code in commit https://github.com/GenericMappingTools/pygmt/pull/2550/commits/5ed896d7e307f759c07de9bdbf197947ee74edc4.
Now the three beachballs are equally spaced; before there was a slightly larger horizontal space between the first and the second beachball.

File: examples/tutorials/advanced/focal_mechanisms.py (hunk 18 above)
Comment (yvonnefroehlich) [reply examples/tutorials/advanced/focal_mechanisms.py]: Oh, yeah! Added in commit https://github.com/GenericMappingTools/pygmt/pull/2550/commits/379cdb1137ce6d43b89e42408d870fabf9bed357.

File: examples/tutorials/advanced/focal_mechanisms.py [hunk 20]
```diff
@@ -0,0 +1,347 @@
...
+fig.basemap(region=region, projection=projection, frame=frame)
+
+fig.meca(
+    spec=aki_single,
+    scale="1c",
+    longitude=-2,
+    latitude=0,
+    depth=0,
+    # Use a 1-point thick, darkorange and solid line
+    pen="1p,darkorange,solid",
```
Comment (seisman) [inline examples/tutorials/advanced/focal_mechanisms.py:161]: The default line style is `solid`, so no need to specify it here:
```suggestion
    pen="1p,darkorange",
```

File: examples/tutorials/advanced/focal_mechanisms.py [hunk 21]
```diff
@@ -0,0 +1,347 @@
...
+)
+
+fig.meca(
+    spec=aki_single,
+    scale="1c",
+    longitude=2,
+    latitude=0,
+    depth=0,
+    outline="1p,darkorange,solid",
```
Comment (seisman) [inline examples/tutorials/advanced/focal_mechanisms.py:170]:
```suggestion
    outline="1p,darkorange",
```

File: examples/tutorials/advanced/focal_mechanisms.py [hunk 22]
```diff
@@ -0,0 +1,347 @@
...
+fig = pygmt.Figure()
+fig.basemap(region=region, projection=projection, frame=frame)
+
+fig.meca(
+    spec=aki_single,
+    scale="1c",
+    longitude=-2,
+    latitude=0,
+    depth=0,
+    nodal="0/1p,black,solid",
```
Comment (seisman) [inline examples/tutorials/advanced/focal_mechanisms.py:195]:
```suggestion
    nodal="0/1p,black",
```

File: examples/tutorials/advanced/focal_mechanisms.py [hunk 23]
```diff
@@ -0,0 +1,347 @@
...
+)
+
+fig.meca(
+    spec=aki_single,
+    scale="1c",
+    longitude=2,
+    latitude=0,
+    depth=0,
+    compressionfill="lightorange",
+    outline="0.5p,black,solid",
```
Comment (seisman) [inline examples/tutorials/advanced/focal_mechanisms.py:205]:
```suggestion
    outline="0.5p,black",
```

File: examples/tutorials/advanced/focal_mechanisms.py [hunk 24]
```diff
@@ -0,0 +1,347 @@
...
+)
+fig.meca(
+    spec=aki_single,
+    scale="1c",
+    longitude=2,
+    latitude=0,
+    depth=0,
+    nodal="1/1p,darkorange,solid",
```
Comment (seisman) [inline examples/tutorials/advanced/focal_mechanisms.py:213]:
```suggestion
    nodal="1/1p,darkorange",
```

File: examples/tutorials/advanced/focal_mechanisms.py [hunk 25]
```diff
@@ -0,0 +1,347 @@
...
+)
+fig.meca(
+    spec=aki_single,
+    scale="1c",
+    longitude=2,
+    latitude=0,
+    depth=0,
+    compressionfill="white@100",
+    extensionfill="white@100",
+    pen="1p,gray30,solid",
```
Comment (seisman) [inline examples/tutorials/advanced/focal_mechanisms.py:223]:
```suggestion
    pen="1p,gray30",
```

File: examples/tutorials/advanced/focal_mechanisms.py [hunk 26]
```diff
@@ -0,0 +1,347 @@
...
+#
+# Required parameters are ``spec`` and ``scale`` as well as ``longitude``, ``latitude``
+# (event location), and depth (if these values are not included in the argument passed
+# to ``spec``). Additionally the ``convention`` parameter is required if ``spec`` is
+# an 1-D or 2-D numpy array; for the input types dictionary and ``pandas.Dataframe``,
+# the focal mechanism convention is automatically determined from dictionary keys or
+# :class:`pandas.DataFrame` column names. The ``scale`` parameter controls the radius
+# of the beachball. By default, the value defines the size for a magnitude of 5 (i.e.,
+# a scalar seismic moment of M0 = 4.0E23 dyn cm). Appand ``"+l"`` to force the radius
+# to be proportional to the seismic moment.
```
Comment (seisman) [inline examples/tutorials/advanced/focal_mechanisms.py:71]:
```suggestion
# of the beachball. By default, the value defines the size for a magnitude of 5 (i.e.,
# a scalar seismic moment of :math:`M_0 = 4.0 \times 10^2`: dyn cm) and the beachball 
# size is proportional to the magnitude. Appand ``"+l"`` to force the radius
# to be proportional to the seismic moment.
```

File: examples/tutorials/advanced/focal_mechanisms.py [hunk 27]
```diff
@@ -0,0 +1,347 @@
...
+for component in ["full", "dc", "deviatoric"]:
+    fig.meca(
+        spec=mt_single,
+        scale="1c",
+        longitude=longitude,
+        latitude=0,
+        depth=0,
+        component=component,
+    )
+    longitude = longitude + 3
```
Comment (seisman) [inline examples/tutorials/advanced/focal_mechanisms.py:103]:
```suggestion
for component, longitude in zip(["full", "dc", "deviatoric"], [-3, 0, 3], strict=True):
    fig.meca(
        spec=mt_single,
        scale="1c",
        longitude=longitude,
        latitude=0,
        depth=0,
        component=component,
    )
```

**Comment by seisman** [comment]
@yvonnefroehlich The tutorial already looks good to me. Are there more changes you'd like to make?

File: examples/tutorials/advanced/focal_mechanisms.py [hunk 28]
```diff
@@ -0,0 +1,346 @@
...
+
+This tutorial focus on how to adjust the display of the beachballs:
+
+- Plotting the components of a seismic moment tensor
+- Filling the quadrants
+- Adjusting the outlines
+- Highlighting the nodal planes
+- Adding offset from the event location
+- Adding a label
+- Using size-coding and color-coding
```
[the remark this answers is not in the document]
Comment (yvonnefroehlich) [reply examples/tutorials/advanced/focal_mechanisms.py]: I am OK with this change as this is only a plain list without links to the single sections.

**Comment by yvonnefroehlich** [comment]
> @yvonnefroehlich The tutorial already looks good to me. Are there more changes you'd like to make?

From the content side I do not  have things in mind I want to change at the moment (actually, the tutorial is already a bit longer). If we have additional GMT flags wrapped (e.g., #3526) we may want consider to extend this tutorial.

File: examples/tutorials/advanced/focal_mechanisms.py [hunk 29]
```diff
@@ -0,0 +1,335 @@
...
+fig.meca(
+    spec=aki_single,
+    scale="1c",
+    longitude=2,
+    latitude=0,
+    depth=0,
+    compressionfill="white@100",
+    extensionfill="white@100",
+    pen="1p,gray30",
+)
```
Comment (seisman) [inline examples/tutorials/advanced/focal_mechanisms.py:212]: Maybe combine the three `Figure.meca` calls into a for-loop and add explanations?
```suggestion
# Plotting the same beachball three times with different settings.
for kwargs in [
    {"compressionfill": "lightorange", "outline": "0.5p,black"},
    {"nodal": "1/1p,darkorange"},
    {"compressionfill": "-", "extensionfill": "-", "pen": "1p,gray30"},
]: 
    fig.meca(
        spec=aki_single,
        scale="1c",
        longitude=0,
        latitude=0,
        depth=0,
        **kwargs,
    )
```

File: examples/tutorials/advanced/focal_mechanisms.py [hunk 30]
```diff
@@ -0,0 +1,335 @@
...
+    nodal="1/1p,darkorange",
+)
+fig.meca(
+    spec=aki_single,
+    scale="1c",
+    longitude=2,
+    latitude=0,
+    depth=0,
+    compressionfill="white@100",
+    extensionfill="white@100",
```
Comment (seisman) [inline examples/tutorials/advanced/focal_mechanisms.py:210]: As mentioned the suggstions below, it's possible to use `"-"` to mean no filling.

File: examples/tutorials/advanced/focal_mechanisms.py [hunk 31]
```diff
@@ -0,0 +1,335 @@
...
+#
+# A moment tensor can be decomposed into isotropic and deviatoric parts. The deviatoric
+# part can be further decomposed into multiple parts (e.g., a double couple (DC) and a
+# compensated linear vector dipole (CLVD)). Use the ``component`` parameter to specify
+# the component you want to plot.
+
+fig = pygmt.Figure()
+fig.basemap(region=region, projection=projection, frame=frame)
+
+for component, longitude in zip(["full", "dc", "deviatoric"], [-3, 0, 2], strict=True):
```
[the remark this answers is not in the document]
Comment (seisman) [reply examples/tutorials/advanced/focal_mechanisms.py:83]: ping @yvonnefroehlich

File: examples/tutorials/advanced/focal_mechanisms.py [hunk 32]
```diff
@@ -0,0 +1,327 @@
...
+Focal mechanisms can be plotted as beachballs with the :meth:`pygmt.Figure.meca` method.
+
+The focal mechanism data or parameters can be provided as various input types: ASCII
+file, :class:`numpy.array`, dictionary, or :class:`pandas.Dataframe`. Different
+conventions to define the focal mechanism are supported: Aki and Richards (``"aki"``),
+global CMT (``"gcmt"``), moment tensor (``"mt"``), partial focal mechanism
+(``"partial"``), and, principal axis (``"principal_axis"``). Please refer to the table
+in the documentation of :meth:`pygmt.Figure.meca` regarding how to set up the input data
+in respect to the chosen input type and convention (i.e., the expected column order,
+keys, or column names). This tutorial we focus on how to adjust the display of the
```
Comment (michaelgrund) [inline examples/tutorials/advanced/focal_mechanisms.py:14]:
```suggestion
keys, or column names). In this tutorial we focus on how to adjust the display of the
```

File: examples/tutorials/advanced/focal_mechanisms.py [hunk 33]
```diff
@@ -0,0 +1,327 @@
...
+aki_single = {"strike": 318, "dip": 89, "rake": -179, "magnitude": 7.75}
+
+
+# %%
+# Plotting a single beachball
+# ---------------------------
+#
+# Required parameters are ``spec`` and ``scale`` as well as ``longitude``, ``latitude``
+# (event location), and depth (if these values are not included in the argument passed
+# to ``spec``). Additionally the ``convention`` parameter is required if ``spec`` is
```
Comment (michaelgrund) [inline examples/tutorials/advanced/focal_mechanisms.py:55]:
```suggestion
# to ``spec``). Additionally, the ``convention`` parameter is required if ``spec`` is
```

File: examples/tutorials/advanced/focal_mechanisms.py [hunk 34]
```diff
@@ -0,0 +1,327 @@
...
+
+
+# %%
+# Highlighting the nodal planes
+# -----------------------------
+#
+# Use the parameter ``nodal`` to highlight specific nodal planes. ``"0"`` refers to
+# both, ``"1"`` to the first, and ``"2"`` to the second nodal plane(s). Only the
+# circumference and the specified nodal plane(s) are plotted, i.e. the quadrants
+# remain unfilled (transparent). We can make usage of the stacking concept of (Py)GMT,
```
Comment (michaelgrund) [inline examples/tutorials/advanced/focal_mechanisms.py:172]:
```suggestion
# remain unfilled (transparent). We can make use of the stacking concept of (Py)GMT,
```

File: examples/tutorials/advanced/focal_mechanisms.py [hunk 35]
```diff
@@ -0,0 +1,327 @@
...
+# and use ``nodal`` in combination with the ``outline``, ``compressionfill`` /
+# ``extensionfill`` and ``pen``  parameters.
```
Comment (michaelgrund) [inline examples/tutorials/advanced/focal_mechanisms.py:174]:
```suggestion
# ``extensionfill`` and ``pen`` parameters.
```

File: examples/tutorials/advanced/focal_mechanisms.py [hunk 36]
```diff
@@ -0,0 +1,327 @@
...
+    )
+fig.show()
+
+
+# %%
+# Adding offset from event location
+# ---------------------------------
+#
+# Specify the optional parameters ``plot_longitude`` and ``plot_latitude``. If ``spec``
+# is an ASCII file with columns for ``plot_longitude`` and ``plot_latitude`` the
```
Comment (michaelgrund) [inline examples/tutorials/advanced/focal_mechanisms.py:213]:
```suggestion
# is an ASCII file with columns for ``plot_longitude`` and ``plot_latitude``, the
```

File: examples/tutorials/advanced/focal_mechanisms.py [hunk 37]
```diff
@@ -0,0 +1,327 @@
...
+    }
+)
+
+
+# %%
+# Adding a label
+# --------------
+#
+# Use the optional parameter ``event_name`` to add a label near the beachball, e.g.,
+# event name or event date and time. Change the font of the the label text by appending
```
Comment (michaelgrund) [inline examples/tutorials/advanced/focal_mechanisms.py:284]:
```suggestion
# event name or event date and time. Change the font of the label text by appending
```

File: examples/tutorials/advanced/focal_mechanisms.py [hunk 38]
```diff
@@ -0,0 +1,327 @@
...
+# **+f** and the desired font (size,name,color) to the argument passed to the ``scale``
+# parameter. Additionally, the location of the label relative to the beachball [Default
+# is ``"TC"``, i.e., Top Center]; can be changed by appending **+j** and an offset can
```
Comment (michaelgrund) [inline examples/tutorials/advanced/focal_mechanisms.py:287]:
```suggestion
# is ``"TC"``, i.e., Top Center] can be changed by appending **+j** and an offset can
```

**Review by michaelgrund — APPROVED** [review]
